### ISSUE TITLE
Excessive cache use with CL=ONE

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylla#3177]
With CL=ONE we spread cache accesses over all three replicas, but to survive a failure we only need to spread the load over two replicas.

**Comment by avikivity** [comment]
I think we need to pick CL replicas out of the first available CL+1 replicas to provide availability in the face of one failure. That translates to 2 from 3 for CL=Q/RF=3 and 1 from 2 for CL=1/RF=3.

**Comment by nyh** [comment]
@avikivity To clarify, by "surviving a failure" do you mean that you want the backup node to also have a fairly warm cache (not merely "survive"), which is why you want to send read requests to two nodes and not to just one?

By the way, something which wasn't clear to me on first read (but should have been...) was the the "first available CL+1 replicas" is calculated per token. So overall, we are still using all nodes for read requests, not just a subset, but each node will be requested (and will cache) fewer vnode ranges. Nice idea.

**Comment by avikivity** [comment]
@nyh correct, survive without loss of quality of service. And yes, all other nodes will be loaded equally.

**Comment by slivne** [comment]
We will need to modify clients for this ....

RoundRobin TokenAware does not have the notion of the "first two" replicas
for CL=ONE - all the queries going to the "third" replica will have a
chance to get higher latencies because of the extra (forced) hop.

How will we handle the case we are querying the same data once with CL=ONE
and once with CL=QUERY.

**Comment by avikivity** [comment]
> We will need to modify clients for this ....

Yes.

> How will we handle the case we are querying the same data once with CL=ONE
and once with CL=QUERY.

I don't think we need to do anything special.
